Dockerfiles/midstream/run_tests.py

```diff
@@ -0,0 +1,65 @@
+        self.operator_work_dir = os.getenv('OPERATOR_WORK_DIR',
+                                           "/project/test_operator_work_dir/")
+        self.work_dir = os.getenv('WORK_DIR',
+                                  "/project/output/")
+        self.playbooks_dir = os.getenv('PLAYBOOKS_DIR',
+                                       "/project/operator-test-playbooks/")
+        self.image_to_test = os.getenv('IMAGE_TO_TEST',
+                                       'quay.io/cvpops/test-operator:v1.0-16')
+        self.umoci_bin_path = os.getenv('UMOCI_BIN_PATH',
+                                        '/usr/local/bin/umoci')
+
+    def run_extract_operator_bundle(self):
+        exec_cmd = "ansible-playbook -i localhost, -c local -v operator-test-playbooks/extract-operator-bundle.yml \
+                    -e 'bundle_image={bundle_image}' \
+                    -e 'operator_work_dir={operator_work_dir}' \
+                    -e 'umoci_bin_path={umoci_bin_path}'".format(
+                                                    operator_dir=self.operator_dir,
+                                                    operator_work_dir=self.operator_work_dir,
+                                                    bundle_image=self.image_to_test,
+                                                    umoci_bin_path=self.umoci_bin_path)
+        result = subprocess.run(exec_cmd, shell=True)
+        return result
+
+    def run_validate_operator_bundle(self):
+        exec_cmd = "ansible-playbook -vvv -i localhost, --connection local \
+                    operator-test-playbooks/validate-operator-bundle.yml \
+                    -e 'operator_dir={operator_dir}' \
+                    -e 'operator_work_dir={operator_work_dir}' \
+                    -e 'work_dir={work_dir}'".format(operator_dir=self.operator_dir,
+                                                    operator_work_dir=self.operator_work_dir,
+                                                    work_dir=self.work_dir)
+        result = subprocess.run(exec_cmd, shell=True)
+        return result
+
+    def test_for_extract_and_validate_bundle_image(self):
+        result = self.run_extract_operator_bundle()
+        self.assertEqual(0,
+                         result.returncode,
+                         "extract-operator-bundle.yml failed")
+        result = self.run_validate_operator_bundle()
+        self.assertEqual(0,
+                         result.returncode,
+                         "validate-operator-bundle.yml failed")
+        self.assertTrue(path.exists("/project/output/validation-rc.txt"))
+        self.assertTrue(path.exists("/project/output/validation-output.txt"))
+        self.assertTrue(path.exists("/project/output/validation-version.txt"))
+        with open('/project/output/validation-rc.txt', 'r') as reader:
+            rc = reader.read()
+            self.assertEqual(0, int(rc), "validation of bundle image failed")
```

Comment:
I've just checked logs files from gitlab actions and I realized there is a huge problem with such executions of tests.
We will not display output of validation anywhere. People will not know what to fix.

```
TASK [validate_operator_bundle : Validate the operator bundle manifest and metadata with operator-sdk bundle validate] ***
task path: /project/operator-test-playbooks/roles/validate_operator_bundle/tasks/main.yml:31
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c 'echo ~root && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp `"&& mkdir "` echo /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899 `" && echo ansible-tmp-1618940583.1714268-409-186136952484899="` echo /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899 `" ) && sleep 0'
Using module file /usr/lib/python3.9/site-packages/ansible/modules/commands/command.py
<localhost> PUT /root/.ansible/tmp/ansible-local-333fve6fbk1/tmpz1tunzmb TO /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899/AnsiballZ_command.py
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899/ /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899/AnsiballZ_command.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python3 /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899/AnsiballZ_command.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1618940583.1714268-409-186136952484899/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": true,
    "cmd": "/usr/local/bin/operator-sdk bundle validate  --verbose /project/test_operator_work_dir/ > /project/output//validation-output.txt 2>&1",
```

We will also need to write another negative test which will fail  just in `validate-operator-bundle.yml`

Reply:
Added output of the bundle validation to the `validate_operator_bundle` role. Working on a negative test.

Reply:
Added the negative test case that fails just in the `validate-operator-bundle.yml` role.

Reply:
Thank you for another negative test.

I am not sure whether people will be able to find where is the problem from the output.
But we can refactor the output later. we shoudl unblock midstream folks.

Reply:
Ack, I'll create a story to improve the ansible playbook output in general as this is an issue with all of the existing tests.